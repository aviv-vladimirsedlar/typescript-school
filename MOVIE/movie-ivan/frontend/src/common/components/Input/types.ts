export interface InputProps {
  'data-testid'?: string;
  className?: string;
  classNameInput?: string;
  error?: string;
  label: string;
  name: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  required?: boolean;
  type?: 'text' | 'password' | 'email' | 'number';
  value: string;
}
