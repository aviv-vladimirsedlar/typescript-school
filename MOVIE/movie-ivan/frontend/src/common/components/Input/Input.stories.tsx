import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test';

import Input from '.';

const meta = {
  title: 'Common/Components/Input',
  component: Input,
  parameters: {},
  tags: ['autodocs'],
  argTypes: {
    type: {
      control: {
        type: 'select',
        options: ['text', 'password', 'email'],
      },
    },
    required: { control: 'boolean' },
    error: { control: 'text' },
  },
} satisfies Meta<typeof Input>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: { label: 'Username', name: 'username', value: '', onChange: fn() },
};

export const Required: Story = {
  args: {
    label: 'Password',
    name: 'password',
    required: true,
    type: 'password',
    value: '',
    onChange: fn(),
  },
};

export const WithError: Story = {
  args: { label: 'Username', name: 'username', value: '', error: 'This field is required', onChange: fn() },
};
