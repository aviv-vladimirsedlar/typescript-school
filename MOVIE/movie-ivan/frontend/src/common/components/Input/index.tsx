import clsx from 'clsx';
import React from 'react';

import { InputProps } from './types';

const Input: React.FC<InputProps> = ({
  'data-testid': testId,
  className,
  classNameInput,
  error,
  label,
  name,
  onChange,
  required = false,
  type = 'text',
  value,
}) => (
  <div className={clsx('mb-5', className)}>
    <label
      aria-labelledby={name}
      htmlFor={name}
      className="mb-2 block text-sm font-medium text-gray-900 dark:text-white"
    >
      {label} {required && <span>*</span>}
    </label>
    <input
      data-testid={testId}
      className={clsx('block w-full rounded-lg bg-gray-50/50 p-4 text-sm text-gray-900', classNameInput, {
        'border border-red-500': !!error,
        'border border-gray-200': !error,
      })}
      onChange={onChange}
      required={required}
      type={type}
      value={value}
    />
  </div>
);

export default Input;
