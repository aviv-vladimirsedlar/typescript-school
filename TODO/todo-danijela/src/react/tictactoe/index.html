<!DOCTYPE html>
<html>
<body>
  <div id="root"></div>
</body>
<script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
<script async src="https://ga.jspm.io/npm:es-module-shims@1.7.0/dist/es-module-shims.js"></script>
<script type="text/javascript" src="index.js"/>
</html>
