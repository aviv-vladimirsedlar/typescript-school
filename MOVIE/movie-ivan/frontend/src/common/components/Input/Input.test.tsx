import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';

import '@testing-library/jest-dom';
import Input from '.';

describe('Input component', () => {
  it('renders the input with correct label', () => {
    render(<Input data-testid="username" label="Username" name="username" value="" onChange={() => {}} />);
    expect(screen.getByTestId(/username/i)).toBeInTheDocument();
  });

  it('renders required asterisk when required is true', () => {
    render(<Input data-testid="email" label="Email" name="email" value="" onChange={() => {}} required />);
    expect(screen.getByText('*')).toBeInTheDocument();
  });

  it('applies error styling when error is present', () => {
    render(<Input data-testid="input" label="Email" name="email" value="" onChange={() => {}} error="Invalid email" />);
    const inputElement = screen.getByTestId('input');
    expect(inputElement).toHaveClass('border-red-500');
  });

  it('calls onChange handler when input value changes', () => {
    const handleChange = jest.fn();
    render(<Input data-testid="input" label="Password" name="password" value="" onChange={handleChange} />);
    const inputElement = screen.getByTestId('input');
    fireEvent.change(inputElement, { target: { value: 'new value' } });
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it('renders the correct initial value', () => {
    render(<Input data-testid="input" label="Search" name="search" value="initial value" onChange={() => {}} />);
    const inputElement = screen.getByTestId('input') as HTMLInputElement;
    expect(inputElement.value).toBe('initial value');
  });
});
