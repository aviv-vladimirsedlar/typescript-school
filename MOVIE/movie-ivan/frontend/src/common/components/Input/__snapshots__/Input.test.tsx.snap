// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should match snapshot when initially rendered 1`] = `
<DocumentFragment>
  <div
    class="mb-5"
  >
    <label
      aria-labelledby="username"
      class="mb-2 block text-sm font-medium text-gray-900 dark:text-white"
      for="username"
    >
      Username 
    </label>
    <input
      class="block w-full rounded-lg bg-gray-50/50 p-4 text-sm text-gray-900 border border-gray-200"
      data-testid="username"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
